tests/ludwig/schema/test_config_object.py

```diff
@@ -0,0 +1,373 @@
+                "name": "text_feature",
+                "type": "text",
+                "preprocessing": {
+                    "missing_value_strategy": "drop_row",
+                },
+                "encoder": {
+                    "type": "rnn",
+                    "bidirectional": True,
+                    "representation": "dense",
+                    "num_layers": 2,
+                },
+            },
+            {
+                "name": "image_feature_1",
+                "type": "image",
+                "preprocessing": {
+                    "height": 7,
+                    "width": 7,
+                    "num_channels": 4,
+                },
+                "encoder": {
+                    "type": "resnet",
+                    "num_channels": 4,
+                    "dropout": 0.1,
+                    "resnet_size": 100,
+                },
+            },
+        ],
+        "output_features": [
+            {
+                "name": "category_feature",
+                "type": "category",
+                "top_k": 3,
+                "preprocessing": {
+                    "missing_value_strategy": "bfill",
+                },
+                "decoder": {
+                    "type": "classifier",
+                    "num_classes": 10,
+                    "use_bias": False,
+                },
+            },
+        ],
+        "combiner": {
+            "type": "concat",
+            "output_size": 512,
+            "weights_initializer": "xavier_uniform",
+            "dropout": 0.2,
+        },
+        "trainer": {
+            "epochs": 50,
+            "batch_size": "auto",
+            "optimizer": {
+                "type": "adam",
+                "beta1": 0.8,
+                "beta2": 0.999,
+                "epsilon": 5e-09,
+            },
+        },
+    }
+
+    config_object = ModelConfig.from_dict(config)
+    assert config_object.input_features.text_feature.encoder.type == "rnn"
+    assert config_object.input_features.text_feature.encoder.num_layers == 2
+    assert config_object.input_features.text_feature.preprocessing.missing_value_strategy == "drop_row"
+
+    assert config_object.defaults.text.encoder.type != "rnn"
+    assert config_object.defaults.text.preprocessing.missing_value_strategy != "drop_row"
+
+    assert config_object.output_features.category_feature.decoder.num_classes == 10
+    assert config_object.output_features.category_feature.top_k == 3
+
+    assert config_object.combiner.output_size == 512
+    assert config_object.combiner.weights_initializer == "xavier_uniform"
+    assert config_object.combiner.fc_layers is None
+
+    assert config_object.trainer.epochs == 50
+    assert config_object.trainer.batch_size == "auto"
+
+    assert config_object.trainer.optimizer.type == "adam"
+    assert config_object.trainer.optimizer.beta1 == 0.8
+    assert config_object.trainer.optimizer.beta2 == 0.999
+    assert config_object.trainer.optimizer.epsilon == 5e-09
```

Comment:
I think this actually may not be the best idea. Funny enough what you're suggesting is actually the original way I was running this test, however, because `to_dict()` returns a fully populated config dictionary, it ends up being way more lines (was like 300-400 lines or something originally). So i figured it would actually be more readable to just check that all the parameters that I am setting in this config dict are set in the config object as opposed to comparing a full config dict. WDYT?

Reply:
Makes sense. Let's go with what you have now.